tests/test_x509.py

```diff
@@ -486,19 +487,41 @@ def test_load_pem_cert(self, backend):
             backend
         )
         assert isinstance(cert, x509.Certificate)
-        assert cert.serial == 11559813051657483483
+        assert cert.serial_number == 11559813051657483483
         fingerprint = binascii.hexlify(cert.fingerprint(hashes.SHA1()))
         assert fingerprint == b"2b619ed04bfc9c3b08eb677d272192286a0947a8"
         assert isinstance(cert.signature_hash_algorithm, hashes.SHA1)
 
+    def test_cert_serial_number(self, backend):
+        cert = _load_cert(
+            os.path.join("x509", "PKITS_data", "certs", "GoodCACert.crt"),
+            x509.load_der_x509_certificate,
+            backend
+        )
+
+        warnings.simplefilter("always", utils.DeprecatedIn10)
+        assert cert.serial == 2
+        assert cert.serial_number == 2
+
+    def test_cert_serial_warning(self, backend):
+        cert = _load_cert(
+            os.path.join("x509", "PKITS_data", "certs", "GoodCACert.crt"),
+            x509.load_der_x509_certificate,
+            backend
+        )
+
+        warnings.simplefilter("always", utils.DeprecatedIn10)
```

Comment:
mmm, so I think you want to wrap this all in a `with warnings.catch_warning` to make sure the filter is restored: https://docs.python.org/2.7/library/warnings.html#warnings.catch_warnings